discovery.go

```diff
@@ -0,0 +1,200 @@
+package main
```

Comment:
Should be `package main` instead. Some tests are currently being skipped due to the package mismatch. As a result, I see a panic that's being overlooked (`NodeSource` is never set) since Travis isn't running the tests.